compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -2875,27 +2890,61 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             hir::TyKind::BareFn(bf) => {
                 require_c_abi_if_c_variadic(tcx, bf.decl, bf.abi, ast_ty.span);
 
-                tcx.mk_fn_ptr(self.ty_of_fn(
+                let fn_ptr_ty = tcx.mk_fn_ptr(self.ty_of_fn(
                     ast_ty.hir_id,
                     bf.unsafety,
                     bf.abi,
                     bf.decl,
                     None,
                     Some(ast_ty),
-                ))
+                ));
+
+                if let Some(bad_param) = bf.generic_params.iter().find(|param| {
+                    matches!(
+                        param.kind,
+                        hir::GenericParamKind::Type { .. } | hir::GenericParamKind::Const { .. }
+                    )
```

Comment:
```suggestion
                    param.kind.is_ty_or_const()
```

Reply:
That's not defined on `hir::GenericParamKind`, just `ty::GenericParamDefKind` and `hir::GenericArg`. I worked around it though.